### DISCUSSION TITLE
Should dotnet-fm runner automatically detect ConventionSets?

### DISCUSSION THREAD

**Discussion by ToddRopog** [fluentmigrator/fluentmigrator#1636] [Q&A]
I have two projects: a DLL which contains the migration classes, and a console app which runs the migrations. The DLL also has a ConventionSet to do some custom naming. The custom naming works correctly when my runner executes the migrations. However, if I use the dotnet-fm runner to execute the migrations, the custom naming does not get applied. Should the dotnet-fm runner see the ConventionSet automatically? [This issue](https://github.com/fluentmigrator/fluentmigrator/issues/916) seems to indicate a change was made so it would work.

**Comment by jzabroski** [comment]
What arguments are you passing in?

**Reply by ToddRopog** [reply]
migrate, --processor, --assembly, and --connection 

Just to be clear, the migrations run successfully, but the ConventionSet is not applied.

**Reply by jzabroski** [reply]
I understand that. Will look.

**Comment by jzabroski** [comment]
This is probably at least a feature request to extend the `list` command so that `list migrations` and `list conventions` (new) are independently verifiable.

Will look later.

**Reply by ToddRopog** [reply]
Just doing a quick check, it looks to me like MigratorConsole.ExecuteMigrations is always loading a DefaultConventionSet.
https://github.com/fluentmigrator/fluentmigrator/blob/master/src/FluentMigrator.Console/MigratorConsole.cs#L394
Maybe that gets overridden later?
I was expecting to see a call to .ScanIn(_[Assembly]_).For.ConventionSet(), which is how my custom runner loads the ConventionSet.

**Reply by jzabroski** [reply]
The point is that there is no command to trace/inspect and display a what if for conventions the same way there is for migrations. 

Migrator.Console is the old source, which uses pre-".NET Core Global Tool" nuget package as "tool" packaging semantics, dating back to Nuget V1 spec.  The dotnet-fm tool is in FluentMigrator.DotNet.Cli
